### PR TITLE
Bugfix/detect baudrate

### PR DESCRIPTION
**Pull request by Jeroen88**
Expose uartStartDetectBaudrate(uart_t *) in esp32-hal-uart.h and call it from HardwareSerial::begin() if baudrate detection is requested (by passing a baudrate of 0) to solve baudrate detection problems.

### CODE REVIEW

File: cores/esp32/esp32-hal-uart.c [hunk 1]
```diff
@@ -401,7 +401,12 @@ uint32_t uartGetBaudRate(uart_t* uart)
     if(uart == NULL) {
         return 0;
     }
+
     uint32_t clk_div = (uart->dev->clk_div.div_int << 4) | (uart->dev->clk_div.div_frag & 0x0F);
+    if(!clk_div) {
```
Comment (me-no-dev) [inline cores/esp32/esp32-hal-uart.c]: in what case is `clk_div` == 0?

Comment (Jeroen88) [reply cores/esp32/esp32-hal-uart.c]: I do not know, but I got a division by zero error and stack trace pointed to uartGetBaudRate(). It was a 'random' error, not reproducible...

Comment (me-no-dev) [reply cores/esp32/esp32-hal-uart.c]: Interesting... ok :)

**Comment by me-no-dev** [comment]
After I merge one request, will update the branch and merge :)

**Comment by Jeroen88** [comment]
@me-no-dev Thnx :)

**Pull request merged**